safenode/src/network/mod.rs

```diff
@@ -0,0 +1,152 @@
+/// Also keeps track of the pending queries/requests and their channels. Once we recieve an event
+/// that is the outcome of a previously executed cmd, send a response to them via the stored channel.
+pub struct EventLoop {
+    swarm: Swarm<SafeNodeBehaviour>,
+    command_receiver: mpsc::Receiver<CmdToSwarm>,
+    event_sender: mpsc::Sender<NetworkEvent>,
+    pending_dial: HashMap<PeerId, oneshot::Sender<Result<()>>>,
+    pending_start_providing: HashMap<QueryId, oneshot::Sender<Result<()>>>,
+    pending_get_providers: HashMap<QueryId, oneshot::Sender<HashSet<PeerId>>>,
+    pending_safe_requests: HashMap<RequestId, oneshot::Sender<Result<SafeResponse>>>,
+}
+
+impl EventLoop {
+    /// Creates the network components
+    /// - The `NetworkApi` to interact with the network layer from anywhere
+    ///   within your application.
+    ///
+    /// - The `NetworkEvent` receiver to get the events from the network layer.
+    ///
+    /// - The `EventLoop` taht drives the network.
+    pub fn new(// secret_key_seed: Option<u8>,
+    ) -> Result<(NetworkApi, impl Stream<Item = NetworkEvent>, EventLoop)> {
+        // Create a random key for ourselves.
+        let keypair = identity::Keypair::generate_ed25519();
+        let local_peer_id = PeerId::from(keypair.public());
+
+        // QUIC configuration
+        let quic_config = libp2p_quic::Config::new(&keypair);
+        let transport = libp2p_quic::async_std::Transport::new(quic_config);
+        let transport = transport
+            .map(|(peer_id, muxer), _| (peer_id, StreamMuxerBox::new(muxer)))
+            .boxed();
+        // Create a Kademlia instance and connect to the network address.
+        // Create a swarm to manage peers and events.
+        let swarm = {
+            // Create a Kademlia behaviour.
+            let mut cfg = KademliaConfig::default();
+            let _ = cfg.set_query_timeout(Duration::from_secs(5 * 60));
+            let kademlia =
+                Kademlia::with_config(local_peer_id, MemoryStore::new(local_peer_id), cfg);
+            let mdns = mdns::async_io::Behaviour::new(mdns::Config::default(), local_peer_id)?;
+            let behaviour = SafeNodeBehaviour {
+                request_response: request_response::Behaviour::new(
+                    SafeMsgCodec(),
+                    iter::once((SafeMsgProtocol(), ProtocolSupport::Full)),
+                    Default::default(),
+                ),
+                kademlia,
+                mdns,
+            };
+
+            let mut swarm =
+                SwarmBuilder::with_async_std_executor(transport, behaviour, local_peer_id).build();
+
+            // // Listen on all interfaces and whatever port the OS assigns.
+            let addr = "/ip4/0.0.0.0/udp/0/quic-v1".parse().expect("addr okay");
+            let _listener_id = swarm.listen_on(addr).expect("listening failed");
+
+            swarm
+        };
+
+        let (command_sender, command_receiver) = mpsc::channel(0);
```

Comment:
To keep the stringency line previously mentioned: `command_sender` => `cmd_sender`